Using rtools to test precomputed bindings on Windows

Ilia-Kosenkov opened the pull request:
After discussing this issue in https://github.com/extendr/rextendr/issues/19, a solution was found to build & test `libR-sys` using only rtools. This does not work for bindings generation (no clang provided with rtools), but is suitable for user consumption.

Currently, rtools mingw64 should be "fixed" (see https://github.com/r-windows/rtools-packages/blob/master/mingw-w64-gcc/PKGBUILD#L305-L309) by duplicating `libgcc.a` as `libgcc_eh.a`, which satisfies `rustc` and linker.

Build & test were reordered. 
First, rtools is configured and precomputed bindings are tested.
Second, msys2 is enabled and bindings generation is tested using msys2.
This does not affect other systems.

Partially addresses #21.

clauswilke commented:
I think it's good to know that this works, but I'm not sure I would merge this at this time. We should just reach out to the Rtools maintainers and ask them to include this fix.

I wouldn't want to tell our endusers that they need to manually copy library files (it will look scary to them) and I think our GH Actions should mirror the installation protocol we recommend in our manuals.

Ilia-Kosenkov commented:
@clauswilke,
Ok but I suggest to leave it open for reference purposes for now.

clauswilke commented:
I'd also be Ok with adding one runner that does it this way, but without deleting the msys2-based runners. This way we can keep an eye on both options until Rtools is fixed.

clauswilke commented on .github/workflows/test.yml:
```diff
@@ -135,10 +135,11 @@ jobs:
-            echo "Building for target: $target"
+            echo "::group::Building for target: $target"
             cargo build -vv --features use-bindgen $(if ($target -ne 'default') {"--target=$target"} )
+            echo "::endgroup::"
```
Doesn't the endgroup statement have to come after the `if` statement checking for errors?

Ilia-Kosenkov replied:
I put "groups" around `cargo` commands so log under spoiler contains only build/test related output.
If `if` condition is met and error is thrown, it will be visible separtely.

clauswilke replied:
Yes, but see: https://github.com/extendr/extendr/pull/130#discussion_r558802290
Does this not apply here?

Ilia-Kosenkov replied:
Oh, I see, that is indeed very stupid of me. I will fix it with the next commit.

Ilia-Kosenkov commented:
I added a copy of the workflow with a reduced number of steps to test rtools build as a separate job, restored the original workflow and added `"::group::"` statements to improve logging of exceptionally long verbose `cargo` outputs.

clauswilke commented on .github/workflows/test.yml:
```diff
@@ -190,13 +192,105 @@ jobs:
+        if: runner.os == 'Windows'
+        # 1. Configure path to libclang
+        # 2. Add path to mingw32/mingw64 -- otherwise library is linked to rtools
+        # 3. Add path to R's i386/x64  -- to solve x86 build/test issue
```
Sorry, missed this earlier: The comments look like they need updating.

clauswilke commented on .github/workflows/test.yml:
```diff
@@ -190,13 +192,105 @@ jobs:
+          }
+          
+
+      # Run tests again using different bindings
```
Same here: comment looks wrong.
